index.d.ts

```diff
@@ -404,3 +404,41 @@ export function stringifyUrl(
 	object: UrlObject,
 	options?: StringifyOptions
 ): string;
+
+/**
+Filter query parameters from a URL.
+
+@param url - The URL containing the query parameters to filter.
+@param keysToKeep - The names of the query parameters to keep. All other query parameters will be removed from the URL.
+@param filter - A filter predicate that will be provided the name of each query parameter and its value. The `parseNumbers` and `parseBooleans` options also affect `value`.
+@returns The URL with the specified query parameters removed.
+
+@example
+```
+queryString.filterElements('https://foo.bar?foo=1&bar=2#hello', ['foo']);
```

Comment:
I think the array version should support exclusions too by doing `['foo', '!bar']`

Reply:
We should pass that to https://github.com/sindresorhus/map-obj

Reply:
If exclusions and inclusions are used in the same array, it presents a different problem. What would `['foo', '!bar']` do vs `['foo']`? Both would produce the same object.

Reply:
Perhaps we could instead add `.filter.exclude()` which does the same as `.filter()` except the other way round.

Reply:
Which is why the original suggestion was to add `.pick()` and `.exclude()` methods, which I still prefer.

Reply:
https://github.com/sindresorhus/filter-obj/issues/9